packages/code-du-travail-api/src/server/routes/annuaire/index.js

```diff
@@ -0,0 +1,66 @@
+const Router = require("koa-router");
+const fetch = require("node-fetch").default;
+const API_BASE_URL = require("../api").BASE_URL;
+
+const elasticsearchClient = require("../../conf/elasticsearch.js");
+const getAnnuaireBody = require("./annuaire.elastic");
+
+const router = new Router({ prefix: API_BASE_URL });
+const index = process.env.ELASTICSEARCH_ANNUAIRE_INDEX || "cdtn_annuaire";
+const routeName = "/annuaire";
+
+// todo @lionelb move to constant file
+const defaultMaxSearchDistance = "30km";
+const adresseApiEndPoint = "https://api-adresse.data.gouv.fr/search";
+
+/**
+ * Return annaire items matching the given address
+ * adress could be a coord or a string.
+ *
+ * @example
+ * http://localhost:1337/api/v1/annuaire/search?coord=lon:lat
+ * http://localhost:1337/api/v1/annuaire/search?q=address to search
+ *
+ * @param {string} :base coord or address for geo search
+ * @returns {Object} Result.
+ */
+router.get(`${routeName}/search`, async ctx => {
+  const { coord, q } = ctx.request.query;
+  const results = searchFromCoord(coord) || (await searchFromQuery(q));
+  if (results) {
+    const [lon, lat] = results;
+    const body = getAnnuaireBody({
+      distance: defaultMaxSearchDistance,
+      coord: { lon, lat }
+    });
+    ctx.body = await elasticsearchClient.search({
+      index,
+      body
+    });
+  } else {
+    ctx.body = { total: 0, hits: { hits: [] } };
+  }
+});
+
+function searchFromCoord(coord) {
```

Comment:
If we want to go full functional `searchFromQuery` should be curried to take the `adresseApiEndPoint` as first arguments

```js
async function searchFromQuery(adresseApiEndPoint){ 
  return query => {
    if (!query) {
      return false;
    }
  // [...]
```

 It's looking very good already :)

Reply:
currying is not needed for now, so I only pass endpoint as a parameter so we could curry it later if we need it